File: 125_algorithms/_exercises/templates/_algorithms_challenges/leetcode/leetCode/Math/223_RectangleArea.cpp
/*
 * @Author: xuezaigds@gmail.com
 * @Last Modified time: 2016-04-12 09:17:10
 */

c.. Solution {
public:
    int computeArea(int A, int B, int C, int D, int E, int F, int G, int H) {
        int left = m..(A, E);
        int bottom = m..(B, F);
        int right = min(C, G);
        int top = min(D, H);
        int size = (D-B) * (C-A) + (H-F) * (G-E);
        __(left < right && bottom < top){
            size -= (top-bottom) * (right-left);
        }
        r_ size;
    }
};

/*
-2
-2
2
2
-2
-2
2
2
0
0
0
0
-1
-1
1
1
*/

File: 125_algorithms/_exercises/templates/_algorithms_challenges/leetcode/leetCode/Math/223_RectangleArea.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-
# @Author: xuezaigds@gmail.com


c.. Solution o..
    ___ computeArea  A, B, C, D, E, F, G, H
        size_1 = (C-A) * (D-B)
        size_2 = (G-E) * (H-F)
        left = m..(A, E)
        bottom = m..(B, F)
        right = m.. C, G)
        top = m.. D, H)

        # There is an area coverd by both the two rectangle
        __ left < right a.. bottom < top:
            r_ size_1 + size_2 - (top-bottom) * (right-left)
        ____
            r_ size_1 + size_2

"""
-2
-2
2
2
-2
-2
2
2
0
0
0
0
-1
-1
1
1
"""
